### PR TITLE
KWS-3070: Implement Play Feeds collection view

### PR DESCRIPTION
**Pull request by Tbruni85**
## JIRA
https://jira.superawesome.tv/browse/KWS-3070
https://jira.superawesome.tv/browse/KWS-3071
https://jira.superawesome.tv/browse/KWS-3072
https://jira.superawesome.tv/browse/KWS-3073

## Description
PlayVideo view implemented

## Screenshots
n/a

### CODE REVIEW

File: PopJamUI/Classes/FeedView/DataProvider/DiscoveryBaseFeedDataProvider.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+//  Created by Tiziano Bruni on 15/06/2018.
+//
+
+import Foundation
+import SAProtobufs
+
+public class DiscoveryBaseFeedDataProvider: BaseFeedDataProvider {
+    
+    override public func fetchData() {
+        if let discoverService = feedItemsService as? DiscoveryServiceProtocol {
```
Comment (devgabrielcoman) [inline PopJamUI/Classes/FeedView/DataProvider/DiscoveryBaseFeedDataProvider.swift]: This `pattern` is kind of anti-polimorphic in my opinion.

See no point in having `BaseFeedDataProvider` get injected with a generic `ServiceProtocol` and then each sub-class has to guard against cases where that service isn't what it needs to be. 
Why bother with polymorphism if you're going to essentially do a big if / else in each subclass?

Rather - try doing this:

BaseFeedDataProvider gets injected with `UserSocialActionsServiceProtocol` & other variables it needs to just trigger "like feed item" actions, get feed item by id & other common things you'll need.

Then `DisocveryBaseFeedDataProvider` can get initialised with the service it needs (`DiscoverServiceProtocol`) and over-write the `fetchData` method in its way.
`HomeFeedDataProvider`, `ProfileFeedDataProvider`, `VideoFeedDataProvider`, etc, all will get initialised with `FeedServiceProtocol`, cause that's the service that they need.

Basically you'll go from a system where you have to have **implicit** knowledge that the `DiscoveryBaseFeedDataProvider` **needs** it's feedItemService instance to be of type `DiscoveryServiceProtocol` to having **explicit** knowledge of that thing 👍

Comment (Tbruni85) [reply PopJamUI/Classes/FeedView/DataProvider/DiscoveryBaseFeedDataProvider.swift]: Yes, I agree with you, we had to patch this for the demo, an update and fixed, version is coming with Myles' PR 3149.
